agent/websockets/connection.go

```diff
@@ -252,3 +264,47 @@ func (conn *Connection) ReadServerMessages() ([]interface{}, error) {
 		return nil, nil
 	}
 }
+
+// injectWebsocketMessage injects a shim header value into a single websocket message in-place.
+// Returns a pointer to a new copy of the struct on success.
+func injectWebsocketMessage(msg *message, injectionPath []string, injectionValues map[string]string) (*message, error) {
+	if msg == nil || injectionValues == nil {
+		return nil, errors.New("unexpected nil message")
+	}
+	if injectionValues == nil {
```

Comment:
This line and line 271 are actually checking same condition

Reply:
Whoops, meant to remove the first condition, good catch